modules/generator/processor/util/util.go

```diff
@@ -31,19 +32,19 @@ func FindAttributeValue(key string, attributes ...[]*v1_common.KeyValue) (string
 	return "", false
 }
 
-func GetSpanMultiplier(ratioKey string, span *v1.Span) float64 {
-	spanMultiplier := 1.0
+func GetMultiplier(ratioKey string, span *v1.Span, rs *v1_resource.Resource) float64 {
+	multiplier := 1.0
 	if ratioKey != "" {
-		for _, kv := range span.Attributes {
+		for _, kv := range append(span.Attributes, rs.Attributes...) {
 			if kv.Key == ratioKey {
 				v := kv.Value.GetDoubleValue()
 				if v > 0 {
-					spanMultiplier = 1.0 / v
+					multiplier = 1.0 / v
```

Comment:
we should return after we find the appropriate value. no reason to keep iterating. i'm not sure why the original code didn't return.